tests/test-amp-style-sanitizer.php

```diff
@@ -156,6 +148,28 @@ public function get_body_style_attribute_data() {
 					':root:not(#_):not(#_):not(#_):not(#_):not(#_):not(#_):not(#_):not(#_):not(#_):not(#_):not(#_) .amp-wp-ab79d9e{color:purple;}',
 				),
 			),
+
+			'col_with_width_attribute' => array(
+				'<table><colgroup><col width="253"/></colgroup></table>',
+				'<table><colgroup><col class="amp-wp-cbcb5c2"></colgroup></table>',
+				array(
+					':root:not(#_):not(#_):not(#_):not(#_):not(#_) .amp-wp-cbcb5c2{width:253px;}',
+				),
+			),
+
+			'col_with_percent_width_attribute' => array(
+				'<table><colgroup><col width="50%"/></colgroup></table>',
+				'<table><colgroup><col class="amp-wp-cd7753e"></colgroup></table>',
+				array(
+					':root:not(#_):not(#_):not(#_):not(#_):not(#_) .amp-wp-cd7753e{width:50%;}',
+				),
+			),
+
+			'col_with_star_width_attribute' => array(
+				'<table><colgroup><col width="0*"/></colgroup></table>',
+				'<table><colgroup><col width="0*"></colgroup></table>',
+				array(),
+			),
```

Comment:
Let's add one more test with a `col` that has `Let's add a `width=50 style="background-color: red; width: 60px;"`. This will ensure that (1) the inline style overrides any `width` attribute, and (2) the new `width` style is correctly merged with the existing styles. In particular, this shoes that `width:50px;` should be _prepended_ and not _appended_ because we need the `width:60px` to “win” over `width:50px`.